src/webui/webapplication.cpp

```diff
@@ -735,6 +737,28 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocation = request().posts["location"].trimmed();
+
+    // check if location valid
+    if(newLocation.isEmpty() || !QDir(newLocation).exists()) return;
+
+    foreach (const QString &hash, hashes) {
+        BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+        if (torrent) {
+            const QString oldLocation = torrent->savePath();
+            Logger::instance()->addMessage(tr("WebUI Set Location: moving '%1' [infohash=%2], from %3 to %4").arg(torrent->name()).arg(hash).arg(oldLocation).arg(newLocation));
+
+            torrent->move(Utils::Fs::expandPathAbs(newLocation));
```

Comment:
Is this synchronous? If not, how does user know when change location has completed?

Reply:
It's asynchronous, after completely moving file the save_path in [torrent api ](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-Documentation#get-torrent-generic-properties) would be updated with new one. However, the failure case would only be detected through timeout :(